### PR TITLE
Updated type definition HTMLButtonElement to reflect HTML Standard

### PR DESCRIPTION
**Pull request by kiriny**

### CODE REVIEW

**Comment by kiriny** [comment]
@microsoft-github-policy-service agree [company="{your company}"]

File: inputfiles/overridingTypes.jsonc [hunk 1]
```diff
@@ -1705,6 +1705,9 @@
                         "labels": {
                             "name": "labels",
                             "overrideType": "NodeListOf<HTMLLabelElement>"
+                        },
+                        "type": {
+                            "overrideType": "\"submit\" | \"reset\" | \"button\" | undefined"
```
Comment (saschanaz) [inline inputfiles/overridingTypes.jsonc:1710]:
```suggestion
                            "overrideType": "\"submit\" | \"reset\" | \"button\"
```

It cannot be `undefined` as it defaults to `submit` per the spec: https://html.spec.whatwg.org/multipage/form-elements.html#attr-button-type

>The [missing value default](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#missing-value-default) and [invalid value default](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#invalid-value-default) are the [Submit Button](https://html.spec.whatwg.org/multipage/form-elements.html#attr-button-type-submit-state) state.

File: baselines/dom.generated.d.ts [hunk 2]
```diff
@@ -6313,7 +6313,7 @@ interface HTMLButtonElement extends HTMLElement {
     /** Sets or retrieves the name of the object. */
     name: string;
     /** Gets the classification and default behavior of the button. */
-    type: string;
+    type: "submit" | "reset" | "button" | undefined;
```
Comment (saschanaz) [inline baselines/dom.generated.d.ts:6316]:
```suggestion
    type: "submit" | "reset" | "button";
```

**Comment by saschanaz** [comment]
Haha, no problem! LGTM, thanks!

**Pull request merged**